### PR TITLE
cbmc-viewer 2.11 (new formula)

### PR DESCRIPTION
**Pull request by ronakfof**
CBMC Viewer scans the output of CBMC and produces a summary that can
be opened in any web browser to understand and debug CBMC findings.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/cbmc-viewer.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      static int global;
+
+      int main() {
+        int *ptr = malloc(sizeof(int));
+        assert(global > 0);
+        return 0;
+      }
+    EOS
+    system "goto-cc", "-o", "main.goto", testpath/"main.c"
+    system "cbmc", "main.goto", "--trace", "--xml-ui", ">", "cbmc.xml"
```
Comment (ronakfof) [inline Formula/cbmc-viewer.rb:46]: Seems like it's unable to open `main.c` as this fails with error when i try `brew test cbmc-viewer` - 
(It works fine on terminal)
```
cbmc.xml

<?xml version="1.0" encoding="UTF-8"?>
<cprover>
<program>CBMC 5.51.0 (cbmc-5.51.0)</program>
<message type="STATUS-MESSAGE">
  <text>CBMC version 5.51.0 (cbmc-5.51.0) 64-bit x86_64 linux</text>
</message>

<message type="ERROR">
  <text>Failed to open input file '&gt;'</text>
</message>

</cprover>
```

Comment (ronakfof) [reply Formula/cbmc-viewer.rb:46]: When passing files with abs path, getting `failed to execute` error
```
==> Testing cbmc-viewer
==> goto-cc -o /tmp/cbmc-viewer-test-20220224-15911-pmiz7t/main.goto /tmp/cbmc-viewer-test-20220224-15911-pmiz7t/main.c
==> cbmc /tmp/cbmc-viewer-test-20220224-15911-pmiz7t/main.goto --trace --xml-ui > /tmp/cbmc-viewer-test-20220224-15911-pmiz7t/cbmc.xml
Last 15 lines from /home/ronakf/.cache/Homebrew/Logs/cbmc-viewer/test.02.cbmc.xml:
2022-02-24 18:59:44 +0000

cbmc /tmp/cbmc-viewer-test-20220224-15911-pmiz7t/main.goto --trace --xml-ui > /tmp/cbmc-viewer-test-20220224-15911-pmiz7t/cbmc.xml

Error: cbmc-viewer: failed
An exception occurred within a child process:
  BuildError: Failed executing: cbmc /tmp/cbmc-viewer-test-20220224-15911-pmiz7t/main.goto --trace --xml-ui > /tmp/cbmc-viewer-test-20220224-15911-pmiz7t/cbmc.xml
```

File: Formula/cbmc-viewer.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    system "goto-cc", "-o", "main.goto", testpath/"main.c"
+    chmod 0755, "main.goto"
+    shell_output("cbmc main.goto --trace --xml-ui > cbmc.xml", 10)
+    shell_output("cbmc main.goto --cover location --xml-ui > coverage.xml")
+    shell_output("cbmc main.goto --show-properties --xml-ui > property.xml")
+    output = shell_output(
+      "cbmc-viewer --goto main.goto --result cbmc.xml " \
+      "--coverage coverage.xml --property property.xml --srcdir .",
+    )
+    assert_match "ERROR: Skipping reachable function", output
```
Comment (ronakfof) [inline Formula/cbmc-viewer.rb:55]: Assert fails even though the output is same - 
```
==> Testing cbmc-viewer
==> goto-cc -o main.goto /tmp/cbmc-viewer-test-20220224-7970-1bzzrrw/main.c
==> cbmc main.goto --trace --xml-ui > cbmc.xml
==> cbmc main.goto --cover location --xml-ui > coverage.xml
==> cbmc main.goto --show-properties --xml-ui > property.xml
==> cbmc-viewer --goto main.goto --result cbmc.xml --coverage coverage.xml --property property.xml --srcdir .
ERROR: Skipping reachable function with invalid source location: {'function': '', 'lastLine': 5}
Error: cbmc-viewer: failed
An exception occurred within a child process:
  Minitest::Assertion: Expected /ERROR:\ Skipping\ reachable\ function/ to match "".
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/minitest-5.15.0/lib/minitest/assertions.rb:183:in `assert'
...
...
```

Comment (ronakfof) [reply Formula/cbmc-viewer.rb]: Resolved

File: Formula/cbmc-viewer.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    system "goto-cc", "-o", "main.goto", testpath/"main.c"
+    shell_output("cbmc main.goto --trace --xml-ui > cbmc.xml", 10)
+    shell_output("cbmc main.goto --cover location --xml-ui > coverage.xml")
+    shell_output("cbmc main.goto --show-properties --xml-ui > property.xml")
+    output = shell_output(
+      "cbmc-viewer --goto main.goto --result cbmc.xml " \
+      "--coverage coverage.xml --property property.xml --srcdir . 2>&1",
+    )
+    assert_match "ERROR: Skipping reachable function with invalid source location: {'function': '', 'lastLine': 5}", output
+  end
```
Comment (ronakfof) [inline Formula/cbmc-viewer.rb:55]: Is this syntax acceptable or can this be improved?

File: Formula/cbmc-viewer.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    EOS
+
+    system "goto-cc", "-o", "main.goto", testpath/"main.c"
+    shell_output("cbmc main.goto --trace --xml-ui > cbmc.xml", 10)
+    shell_output("cbmc main.goto --cover location --xml-ui > coverage.xml")
+    shell_output("cbmc main.goto --show-properties --xml-ui > property.xml")
+    shell_output(
+      "cbmc-viewer --goto main.goto --result cbmc.xml " \
+      "--coverage coverage.xml --property property.xml --srcdir . 2>&1",
+    )
```
Comment (carlocab) [inline Formula/cbmc-viewer.rb:53]:
```suggestion
    system bin/"goto-cc", "-o", "main.goto", testpath/"main.c"
    (testpath/"cbmc.xml").write shell_output("cbmc main.goto --trace --xml-ui", 10)
    (testpath/"coverage.xml").write shell_output("cbmc main.goto --cover location --xml-ui")
    (testpath/"property.xml").write shell_output("cbmc main.goto --show-properties --xml-ui")
    system bin/"cbmc-viewer", "--goto", "main.goto",
                              "--result", "cbmc.xml",
                              "--coverage", "coverage.xml",
                              "--property", "property.xml",
                              "--srcdir", "."
    assert_predicate testpath/"result/html/index.html", :exist?
```

**Comment by SMillerDev** [comment]
Hi, this seems like a nice software, but not notable enough for us to package in Homebrew for now. This is not a decision on the merits of the software, but we cannot distribute every single piece of software and have to draw a line on “notability” somewhere. Of course, we would be happy to reconsider in the future.

Please consider hosting it in a personal tap, which is very easy to do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

**Pull request closed without merging**